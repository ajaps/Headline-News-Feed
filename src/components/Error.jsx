import PropTypes from 'prop-types';
import React from 'react';

const Error = props => (
    <div>
      <p> The App encoutered an error while fetching data</p><br />
      <p>Ensure you're connected to the internet and try again</p>
      <p> ERROR: {props.error} </p>
    </div>
  );

Error.propTypes = {
  error: PropTypes.string.isRequired,
};

export default Error;
