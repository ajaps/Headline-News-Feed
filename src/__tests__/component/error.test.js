import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Error from '../../components/Error.jsx';


describe('Footer', () => {
  let app;
  beforeEach(() => {
    app = shallow(<Error error={'Failed to fetch'}/>);
  });

  it('should render as expected', () => {
    const tree = toJson(app);
    expect(tree).toMatchSnapshot();
  });

  it('contains the props that was passed "Failed to fetch" ', () => {
    expect(app.exists('Failed to fetch')).toBe(true);
  });
});
