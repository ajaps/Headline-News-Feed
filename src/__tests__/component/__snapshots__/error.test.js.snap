// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component should render as expected 1`] = `
<div>
  <p>
     The App encoutered an error while fetching data
  </p>
  <br />
  <p>
    Ensure you're connected to the internet and try again
  </p>
  <p>
     ERROR: 
    Failed to fetch
     
  </p>
</div>
`;
